### PR TITLE
added advanced http and firewall config options to api

### PR DESCRIPTION
**Pull request by abdulrahmantkhalifa**
added advanced flag in gateway schema to prevent adjustments to the
config through ays after custom config has been uploaded

### CODE REVIEW

File: templates/gateway/actions.py [hunk 1]
```diff
@@ -90,7 +90,8 @@ def init(job):
 
 def install(job):
     # nothing to do here all our children will be created by ays automagic
```
Comment (grimpy) [inline templates/gateway/actions.py]: maybe remove this false comment

File: templates/gateway/config.yaml [hunk 2]
```diff
@@ -6,7 +6,7 @@ links:
     argname: node
   consume:
     - auto: false
-      min: 1
+      min: 0
```
Comment (grimpy) [inline templates/gateway/config.yaml]: why?

File: api/node/node_api_SetGWFWConfig.go [hunk 3]
```diff
@@ -1,10 +1,67 @@
...
+	err = client.Filesystem(containerClient).Upload(r.Body, "/etc/nftables.conf")
+	if err != nil {
+		fmt.Errorf("Error uploading file to container '%s' at path '%s': %+v.\n", gwname, "/etc/nftables.conf", err)
+		tools.WriteError(w, http.StatusInternalServerError, err)
+		return
+	}
+
+	service, res, err := api.AysAPI.Ays.GetServiceByName(gwname, "gateway", api.AysRepo, nil, nil)
+
+	if !tools.HandleAYSResponse(err, res, w, "Getting container service") {
```
Comment (serboctor) [inline api/node/node_api_SetGWFWConfig.go]: Incorrect error message

File: api/node/node_api_SetGWFWConfig.go [hunk 4]
```diff
@@ -1,10 +1,67 @@
...
+		Nics:         gatewayBase.Nics,
+		Httpproxies:  gatewayBase.Httpproxies,
+		Portforwards: gatewayBase.Portforwards,
+		Advanced:     true,
+	}
+
+	obj := make(map[string]interface{})
+	obj[fmt.Sprintf("gateway__%s", gwname)] = gatewayNew
+
+	if _, err := tools.ExecuteBlueprint(api.AysRepo, "gateway", gwname, "", obj); err != nil {
```
Comment (serboctor) [inline api/node/node_api_SetGWFWConfig.go]: action name missing

File: api/node/node_api_SetGWHTTPConfig.go [hunk 5]
```diff
@@ -1,10 +1,67 @@
...
+		Nics:         gatewayBase.Nics,
+		Httpproxies:  gatewayBase.Httpproxies,
+		Portforwards: gatewayBase.Portforwards,
+		Advanced:     true,
+	}
+
+	obj := make(map[string]interface{})
+	obj[fmt.Sprintf("gateway__%s", gwname)] = gatewayNew
+
+	if _, err := tools.ExecuteBlueprint(api.AysRepo, "gateway", gwname, "", obj); err != nil {
```
Comment (serboctor) [inline api/node/node_api_SetGWHTTPConfig.go]: action name missing

**Pull request merged**